core/Db/TransactionLevel.php

```diff
@@ -40,6 +40,10 @@ public function canLikelySetTransactionLevel()
 
     public function setUncommitted()
     {
+        if ($this->db->supportsUncommitted === false) {
```

Comment:
Not sure if I'm seeing this right. Wouldn't this need to be `if ($this->db->supportsUncommitted !== null) {` because otherwise if `supportsUncommitted = true;` then it wouldn't use the cached result? @peterhashair @sgiehl

Reply:
@tsteur right, that make sense, just checking should it be return like that.
```php
        if ($this->db->supportsUncommitted !== null) {
            return $this->db->supportsUncommitted;
        }
```

Reply:
I believe that makes sense. Be good to test it and double check it